api/clroni/clroni/Context.cs

```diff
@@ -246,6 +246,8 @@ public Frame ReadFrame()
             Frame frame;
             int rc = NativeMethods.oni_read_frame(handle, out frame);
             if (rc < 0) { throw new ONIException(rc); }
+            var data_bytes = ((frame_t*)frame.DangerousGetHandle())->data_sz;
+            frame.addMemoryPressure(data_bytes);
```

Comment:
This does not feel right. Maybe oni_read_frame could return an IntPtr and have an overloaded Frame constructor that does this